community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/Operations.java

```diff
@@ -381,33 +388,12 @@ public void relationshipDelete( long relationship )
     }
 
     @Override
-    public boolean nodeAddLabel( long node, int nodeLabel ) throws KernelException
+    public boolean nodeAddLabel( long node, int nodeLabel ) throws EntityNotFoundException
     {
-        assertOpen();
-
-        //TODO when singleNode is txState aware we can remove this extra check
-        if ( ktx.hasTxStateWithChanges() )
-        {
-            boolean justAdded = ktx.txState().nodeIsAddedInThisTx( node );
-            if ( justAdded && ktx.txState().nodeStateLabelDiffSets( node ).isAdded( nodeLabel ) )
-            {
-                //the newly added node already have the label
-                return false;
-            }
-            else if ( justAdded )
-            {
-                // we have a new node, let's add the label
-                ktx.txState().nodeDoAddLabel( nodeLabel, node );
-                updater.onLabelChange( nodeLabel, nodeCursor, propertyCursor, ADDED_LABEL );
-                return true;
-            }
-            if ( ktx.txState().nodeIsDeletedInThisTx( node ) )
-            {
-                // already deleted, false or EntityNotFoundException
-                return false;
-            }
-        }
+        acquireSharedLabelLock( nodeLabel );
+        acquireExclusiveNodeLock( node );
 
+        assertOpen();
```

Comment:
Why not do the `assertOpen` call before we take any locks?

Reply:
Because that is the order we used to do it, e.g. https://github.com/neo4j/neo4j/blob/3.4/community/kernel/src/main/java/org/neo4j/kernel/impl/api/LockingStatementOperations.java#L95
I assumed there was some deep underlying reason I couldn't figure out.